Release RST 5.1

ksterne opened the pull request:
# Release RST 5.1

- [ ] PATCH
- [x] MINOR
- [ ] MAJOR

This is the start of the 5.1 release pull request which I'm going off the template used for the 5.0 release pull request.  Also need to consult the release guidelines/checklist that we made up a few years ago.  Here's the key changes for this release:

- New additions
   - Addition of `AACGM_v2` and IGRF coefficients to support mapping until the beginning of 2030
   - New tool (**`find_elvstat`**) for info on realistic number of elevation angle measurements in a fit-level file
   - New conversion tool (**`oldsndtosnd`**) to convert old sounding data (`snd`) written in binary format to the more modern `dmap` format  
   - New derivation of eccentric dipole coordinates for the TS18 and TS18-Kp convection models
   - Addition of slant range parameter (`srng`) to grid and map files
- New/Updated Radar Information
   - Addition of Iceland East (`ice`), West (`icw`); Longjing East (`lje`), West (`ljw`); Siziwanqi East (`sze`), West (`szw`); Hejing East (`hje`), West (`hjw`) radar information
   - Updated hdw.dat files for Blackstone (`bks`), Kodiak (`kod`)
   - Updated radar location information based on satellite imagery
   - Addition of `tdiff.dat` files for several radars
- Updates to Current Binaries and Libraries
   - Modification to `grid_filter` binary to allow filtering data based on range as well as control program ID.
   - Updates to `fitacfclientgui` to allow viewing range gate or slant range, on-screen key menu, selection of groundscatter flag, and color scale control
   - Addition of `-xshift` and `-yshift` options to `field_plot` binary to allow shifts to plotting area; also addition of `-databeam` option to plot field-of-view edges
   - Addition of `-dot` option in `field_plot` to mark radar location
   - Improvements for plotting `snd` data with `time_plot`
   - Updates to `fit` library to check for valid elevation angle information and use the sky noise from fitted data to populate the `noise` fields in grid and map files
   - Support for beam offset parameter included in `rtsnd` binary
- Documentation, Bug fixes and Miscellaneous
   - Improvements to `xmldoc` and `scdoc` binaries used for documentation creation
   - Updating version of `mpfit` from 1.4 to 1.5
   - Various documentation improvements (e.g. updated installation instructions for Mac OS)
   - Various typo errors, bug fixes, and memory leak fixes
   - Updates to compilation checks in github development environment
   - Support for `.tcsh` profiles has been **removed**



## Deadline: 16 April 2025

Please comment in the Pull Request conversation if you need additional time to test the release.

## Testing

- [ ] Mac OSx
- [x] Linux deb - Debian, Ubuntu, Mint
- [ ] Linux rpm - OpenSuse, Fedora, CentOS

Do any binaries/libraries require special attention during testing?

## Checklist

- [x] Update radar.dat (if required) Check closer to the testing deadline
- [x] Update hardware files (if required) Check closer to the testing deadline
- [x] Update .rst.version
- [x] Update author list in .zenodo.json check again before merging
- [x] Update author list in AUTHORS.md

## Extra Notes
A description of the release will appear soon.

## Approving the release
After testing, please state which operating system you used and what you tested.

If you found no bugs during testing, please approve the PR.

pasha-ponomarenko committed:
Merge pull request #555 from SuperDARN/main

Merging `main` back into `develop`

egthomas committed:
Fixing typo in make_raw documentation

egthomas committed:
Adding Iceland East and West hdw files and updating radar.dat

egthomas committed:
Fixing typo in fitacfclientgui documentation

egthomas committed:
Merge pull request #558 from SuperDARN/iceland_hdw

Adding Iceland East and West hdw files and updating radar.dat

egthomas committed:
Fixing broken HTML documentation links

egthomas committed:
Adding new -databeam option to field_plot to draw radar FOV edges according to beams present in scan

ksterne committed:
Merge pull request #557 from SuperDARN/fix_make_raw_typo

Fixing typo in make_raw and fitacfclientgui documentation

egthomas committed:
Fixing typo in rposdlm causing error when returning site status with RadarYMDHMSGetSite

egthomas committed:
Fixing typo in aacgmdlm

egthomas committed:
Modifying IQEncode to remove extra multiplication by 2 causing bug in trim_grid output data size, and modifying make_raw, make_sim, and sim_real to correctly handle the IQ chnnum

egthomas committed:
Modifying column headers in tdiff.dat.tst for better alignment

ecbland committed:
Merge pull request #566 from SuperDARN/fix_iqdat_chnnum

Fixing bug in trim_iq

egthomas committed:
Updating mpfit library used by lmfit from 1.4 to 1.5

egthomas committed:
Major updates to fitacfclientgui, including: on-screen menu explaining keyboard controls; use of a data buffer allowing all beams to update when plotting options change; a new ground scatter flag option (which can also be toggled with the g key); and up/down arrow control of the color scale maximum

aburrell committed:
ENH: added elevation stats function

Added a function to count the number of elevation angles with potentially valid values in fit-esque files.

aburrell committed:
ENH: added elv stat header

Added a header for elevation angle statistical functions.

aburrell committed:
DOC: added elvstat docs

Created documentation for the elvstat functions.

aburrell committed:
ENH: added elevation stat tool

Created a tool to print the number of potentially valid elevation angles in fit-esque file(s).

aburrell committed:
BUG: free malloced memory

Free the `dnames` pointer once it's no longer used.

aburrell committed:
ENH: added requirements file

Added a requirements file for the CI and ease of installation for users.

aburrell committed:
TST: created a CI job

Added a GitHub Actions job to test compilation on the latest ubuntu system with the current dependencies.

aburrell committed:
BUG: fixed requirements filename

Fixed the requirements filename to be correct.

aburrell committed:
BUG: added CDF download

Added the download of the CDF tarball.

aburrell committed:
BUG: removed tabs

Removed tabs from the yaml file.

aburrell committed:
BUG: changed env variables

Changed the way that environment variables are defined in the test.

aburrell committed:
BUG: GitHub actions syntax

Fixed the syntax for GitHub Action environment variables.

aburrell committed:
BUG: fixed env syntax

Fixed the syntax in the `env` section of the workflow.

aburrell committed:
BUG: another env fix attempt

Another attempt to fix the environment variable definition with a test to see if it's successful.

aburrell committed:
BUG: removed tab

Remove a stray tab.

aburrell committed:
BUG: another env attempt

Yet another env attempt, this time using working-directory.

aburrell committed:
BUG: pwd attempt for RSTPATH

Try using pwd to define the RSTPATH environment variable.

aburrell committed:
TST: remove test line

Remove environment variable test line.

egthomas committed:
Fixing memory error in snd library and binaries when setting combf

egthomas committed:
Fixing memory leak in RadarParmDecode

egthomas committed:
Fixing memory leak in FitDecode

aburrell committed:
Merge pull request #567 from SuperDARN/tdiff_header

Modifying column headers in tdiff.dat file for better alignment

egthomas committed:
Modifying dmaptoncdf error messages (changing CDF to netCDF)

egthomas committed:
Modifying dmaptocdf to print CDF error message on failure

egthomas committed:
Modifying field_plot and MapStereographic function to allow user-controlled shifting of the plotting region via new -xshift and -yshift options

aburrell committed:
STY: rename requirements file

Renamed the requirements file to have a hidden name, decluttering the directory.

aburrell committed:
TST: added doc test

Added a test for `make.doc`.

aburrell committed:
REV: revert commands to include directory change

Compilation tests failed, so reverting changes to the call.

aburrell committed:
Revert "REV: revert commands to include directory change"

This reverts commit b0afff5eaf7ec373b1ee21265f534894aaa31b3a.

aburrell committed:
BUG: fixed requirements call

Fixed the call to the old requirements file.

aburrell committed:
BUG: set env variables

Ensure the environment variables are set in the doc call.

egthomas committed:
Modifying fitacfclientgui to display sky noise rather than deprecated noise.mean parameter

ecbland committed:
Merge pull request #562 from SuperDARN/fix_aacgmdlm_bug

Fixing typo in IDL AACGM DLM

egthomas committed:
Modifying fitacfclientgui so color mode is enabled by default

ksterne committed:
Merge pull request #569 from SuperDARN/update_fitacfclientgui

Updates to fitacfclientgui to make `-color` option default (eliminates the `-color` option) and improved functionality of the GUI.

ecbland committed:
Merge pull request #561 from SuperDARN/fix_rposdlm_bug

Fixing bug in IDL DLM RadarYMDHMSGetSite

billetd committed:
Update mac_install.md

Added instructions for finding the X11 path, as well as clarifying what the directory should look like.

billetd committed:
Update mac_install.md

Added instructions for users with ARM64 architecture Macs. This is required so the shared executable are compiled correctly.

egthomas committed:
Modifying rawacf.md and fitacf.md documentation for more accurate mplgexs description

aburrell committed:
Merge pull request #576 from SuperDARN/fix_memory_leaks

Fix memory leaks

ecbland committed:
Merge pull request #585 from SuperDARN/fix_docs

Modifying documentation for more accurate mplgexs description

egthomas committed:
Modifying scdoc and xmldoc binaries to include option error handling support

egthomas committed:
Modifying make.build to compile xmldoc twice so the hlpstr.h file can be correctly populated and displayed with --help

egthomas committed:
Removing unnecessary phidiff comments from elevation_v2 algorithm

egthomas committed:
Modifying FitDecode of fit library to check against both phi0 and elv fields when reading fitted XCF data (ie real-time elevation angles from older radars that don't transmit all fields in fit structure)

ecbland committed:
Merge pull request #589 from SuperDARN/fix_xmldoc_bins

Fix xmldoc bins

ecbland committed:
Merge pull request #591 from SuperDARN/update_elev_comments

Removing phidiff comments from elevation_v2 algorithm

ecbland committed:
Merge pull request #582 from SuperDARN/mac_doc_fix

Small changes to Mac install instructions.

ecbland committed:
Merge pull request #577 from SuperDARN/fix_dmaptoncdf_errors

Updating dmaptocdf and dmaptoncdf error messages

egthomas committed:
Update codebase/superdarn/src.bin/tk/plot/field_plot.1.12/doc/field_plot.doc.xml

Co-authored-by: Emma Bland <22493623+ecbland@users.noreply.github.com>

egthomas committed:
Modifying rtcfit to fix documentation and remove unused -f option

ecbland committed:
Merge pull request #578 from SuperDARN/feature/field_plot_xyshift

Adding new -xshift and -yshift options to field_plot

ecbland committed:
Merge pull request #592 from SuperDARN/update_fitdecode

Modifying FitDecode of fit library to check against both phi0 and elv

ecbland committed:
Merge pull request #593 from SuperDARN/fix_rtcfit

Modifying rtcfit to fix documentation and remove unused -f option

egthomas committed:
Modifying RadarParmDecode to set ifmode to zero (instead of -1) if the field is not present, eg for QNX4-era (or earlier) radars

egthomas committed:
Fixing array indexing bugs in make_sim and sim_real

egthomas committed:
Updating fitacfclientgui to add support for toggling betwen range gate number and slant range (km) by pressing r key or with -r command line option, and changing to only print origin information if present in real-time data stream

egthomas committed:
Removing tcsh profiles

egthomas committed:
Modifying .profile.bash to remove unnecessary PATH additions

egthomas committed:
Modifying fitacfclientgui to respond more quickly to user input

egthomas committed:
Fixing bug in GridTableWrite

aburrell committed:
STY: moved requirements file

Moved the requirements file to the workflow directory.

aburrell committed:
DOC: added AGB to zenodo

Added my name and affiliation to the zenodo json.  Updated contributions in the Authors file.

aburrell committed:
BUG: moved requirements to better place

Moved the requirements file to a sensible place in the codebase.

aburrell committed:
TST: added a RTD yaml

readthedocs now requires a configuration file.

aburrell committed:
Revert "TST: added a RTD yaml"

This reverts commit 4a31656563a32158b4997f8c82461d7ca62810be.

ecbland committed:
TST: Add .readthedocs.yaml configuration file for mkdocs

egthomas committed:
Fixing bug in time_plot when plotting snd data, where last record before end of minute would be plotted at the wrong time

egthomas committed:
Updating radar.dat and hdw files for increased latitude/longitude precision from commercial satellite imagery, and adding new NSSC radars

egthomas committed:
Fixing out of bounds array access when reading text file in a few binaries

egthomas committed:
Moving dayno function from aacgm_v2 library to general rtime library

egthomas committed:
Adding eccentric dipole coordinate functionality in IGRF library and ecdipeval routine

egthomas committed:
Updating superdarn libraries for eccentric dipole support

egthomas committed:
Updating make_grid, map_grd, map_addhmb, map_fit, and solve_model binaries for eccentric dipole support

egthomas committed:
Modifying map_addmodel to add eccentric dipole support, and also new options to calculate model vectors at all possible grid locations or only where data are available, or to calculate residual velocity vectors between model and observations

egthomas committed:
Adding TS18 model coefficient files for eccentric dipole coordinates

egthomas committed:
Removing leftover comment from igrfdlm.c

egthomas committed:
Modifying field_plot to mark the radar location with a dot using a new -dot command line option

egthomas committed:
Adding oldsndtosnd binary to convert old binary format sounding mode files to modern dmap format snd files

ecbland committed:
Bugfix: Needed to set xqflg=0 to properly remove isolated points

egthomas committed:
Modifying rtsnd to add support for bmoff parameter from hdw file

egthomas committed:
Adding .gitattributes file

pasha-ponomarenko committed:
Merge pull request #612 from SuperDARN/fix/fit_speck_removal

Bugfix: setting `xqflg` to zero in `fit_speck_removal`

egthomas committed:
Modifying FitReadRadarScan to store sky noise in RadarBeam structure rather than search noise (which is not used by all radars)

egthomas committed:
Modifying grid and map file documentation for noise changes, and fixing a few formatting errors

aburrell committed:
Merge pull request #575 from SuperDARN/ci_compilation_test

Added CI compilation test

egthomas committed:
Adding XML / command-line documentation for find_elvstat binary

pasha-ponomarenko committed:
Merge pull request #597 from SuperDARN/update_gui_range

Updating fitacfclientgui to add support for toggling between gate and slant range

pasha-ponomarenko committed:
Merge pull request #602 from SuperDARN/fix_time_plot_snd_bug

Fixing bug in time_plot when plotting snd data

pasha-ponomarenko committed:
Merge pull request #595 from SuperDARN/fix_sim_bugs

Fixing array indexing bugs in make_sim and sim_real

egthomas committed:
Merge pull request #574 from SuperDARN/fit_elv_stat

Elevation statistics tool

pasha-ponomarenko committed:
Merge pull request #606 from SuperDARN/feature/field_plot_radar_dot

Modifying field_plot to mark the radar location with a dot

pasha-ponomarenko committed:
Merge pull request #568 from SuperDARN/update_mpfit

Updating mpfit library used by lmfit from version 1.4 to 1.5

pasha-ponomarenko committed:
Merge pull request #600 from SuperDARN/fix_gtablewrite_bug

Fixing bug in GridTableWrite

pasha-ponomarenko committed:
Merge pull request #614 from SuperDARN/fix_radarscan_noise

Modifying FitReadRadarScan to store sky noise in RadarBeam structure rather than search noise

ksterne committed:
Merge pull request #594 from SuperDARN/fix_ifmode

Modifying RadarParmDecode to set ifmode to zero by default

egthomas committed:
Fixing elvstat library XML documentation

aburrell committed:
Merge pull request #615 from SuperDARN/fix_elvstat_doc

Fixing elvstat library XML documentation

ksterne committed:
Merge pull request #559 from SuperDARN/fix_broken_doc_links

Fixing broken HTML documentation links

pasha-ponomarenko committed:
Merge pull request #605 from SuperDARN/hdw_location_update

Updating radar.dat and hdw files for increased latitude/longitude precision and new NSSC radars

egthomas committed:
Merge branch 'develop' into feature/field_plot_databeam

pasha-ponomarenko committed:
Merge pull request #560 from SuperDARN/feature/field_plot_databeam

Adding new -databeam option to field_plot

egthomas committed:
Initial commit adding new srng field to grid and map files describing the slant range of a grid vector

egthomas committed:
Updating documentation for new srng field in grid and map files

egthomas committed:
Modifying grid_plot to allow plotting slant range as filled cells (similar to power and width)

egthomas committed:
Modifying map_plot to allow plotting slant range as filled cells

egthomas committed:
Modifying grid_filter to exclude grid vectors from a file that do not meet user-defined slant range criteria

egthomas committed:
Modifying grid_filter to optionally exclude grid vectors from a file that do not CPID criteria listed in new cpid.dat file

egthomas committed:
Adding header information and in-line comments to cpid.dat file

egthomas committed:
Updating tdiff.dat files for Thomas et al (2024) multi-frequency calibration results

egthomas committed:
Updating channel numbers in tdiff.dat files

RemingtonRohel committed:
Removed all instances of CMPLX.

* Specified double precision for complex numbers in rylm.c
* Simplified complex number instantiation/multiplication in some places. Previously, some calculations were extracting the real and imaginary parts separately and using them in a CMPLX() call to create a new complex number, rather than simply using the already-complex number.
* Answer at https://stackoverflow.com/a/59597456 explains that CMPLX is missing on MacOS when compiling with gcc because the responsibility of defining macros like CMPLX is not up to gcc.

RemingtonRohel committed:
Removed ``#ifdef DARWIN` blocks from `locate.c` files

aburrell committed:
MAINT: update .gitignore

Update .gitignore to include MacOS hidden directory files.

aburrell committed:
TST: added MacOS compilation tests

Expanded compilation tests to macs.

aburrell committed:
BUG: remove sudo from brew

Homebrew no longer supports sudo.

aburrell committed:
BUG: change brew input format

Try a different method of installing dependencies in brew from a file.

aburrell committed:
BUG: another xargs attempt

Try using xargs another way.

pasha-ponomarenko committed:
Merge pull request #620 from RemingtonRohel/FIX/cmplx_removal

FIX: Bug when `CMPLX` macro undefined.

aburrell committed:
BUG: remove brew tap

Fails to tap wget, also try a more forceful brew update.

aburrell committed:
BUG: updated CDF env

Changed the CDF env variable.  Also separated the linux and mac builds.

aburrell committed:
BUG: switch to gcc

Switch from clang to gcc.

aburrell committed:
BUG: fix OS flag

Fixed a bug in the CDF OS flag.

aburrell committed:
BUG: remove GCC

Try with clang again.

aburrell committed:
BUG: add if to linux build

Added a missing if statement to the linux build set of commands.

aburrell committed:
BUG: update path

Update path so libpng can be found.

aburrell committed:
BUG: remove tab

Remove tab from yaml.

aburrell committed:
BUG: fix syntax

Fix the path syntax.

aburrell committed:
BUG: install xquartz

X11 is not installed by default on GitHub Actions.

aburrell committed:
BUG: add cask flag

Add cask flag to xquartz dependencies.

aburrell committed:
BUG: move xquartz install

Move xquatrz install into the test yaml.

aburrell committed:
BUG: update netcdf path

Update the netcdf path.

aburrell committed:
BUG: remove tab

Remove the tab from the yaml.

aburrell committed:
BUG: create test profile files

Create test profile files for use.

aburrell committed:
BUG: remove tabs

Remove tabs from yaml.

aburrell committed:
BUG: update makelib

Add makelib.darwin to the list of updated files.

RemingtonRohel committed:
Fixed -1 error in cnvmodel.c

egthomas committed:
Modifying trim_iq to add support for optional chnnum to override default value from iq structure

mtwalach committed:
Merge pull request #622 from SuperDARN/develop

Add mac OS fixes to pull request

aburrell committed:
TST: remove debug lines

Remove echo commands that were used for debugging.

mtwalach committed:
Merge pull request #621 from SuperDARN/macos_comp_tests

MacOS Comp Tests

ksterne committed:
Merge pull request #596 from SuperDARN/cleanup_scripts

Cleaning up profile scripts

ksterne committed:
Update after USRP install in March 2024

ksterne committed:
Merge pull request #611 from SuperDARN/add_gitattributes

Adding .gitattributes file

egthomas committed:
Merge pull request #627 from SuperDARN/ehn_bks_hdw_update

Update to hdw.dat.bks

RemingtonRohel committed:
Fix minor typo in fitacf.2.5/noise_acf.c

egthomas committed:
Merge pull request #623 from RemingtonRohel/develop

Fixed -1 error in cnvmodel.c

egthomas committed:
Updating AACGM and IGRF code for new IGRF14 coefficients, and adding native IDL support for eccentric dipole coordinate transformations

egthomas committed:
Merge branch 'develop' into feature/ecdip_and_grid_updates

egthomas committed:
Modifying workflow main.yml to use CDF 39_1

ksterne committed:
Merge pull request #610 from SuperDARN/fix_rtsnd

Modifying rtsnd to add support for bmoff parameter from hdw file

egthomas committed:
Update codebase/superdarn/src.bin/tk/reformat/oldsndtosnd.1.0/doc/oldsndtosnd.doc.xml

Co-authored-by: Kevin Sterne <kevintyler@vt.edu>

ksterne committed:
Merge pull request #608 from SuperDARN/feature/oldsndtosnd

Adding oldsndtosnd binary

ksterne committed:
Merge pull request #607 from SuperDARN/feature/ecdip_and_grid_updates

Eccentric dipole coordinates and IGRF14 updates

egthomas committed:
Adding new TS18-Kp model coefficients rederived for eccentric dipole coordinates

egthomas committed:
Modifying cnvmodel library, map_addmodel, and solve_model to support loading TS18-Kp eccentric dipole model coefficients

egthomas committed:
Modifying workflow main.yml to use CDF 39_0

aburrell committed:
Merge pull request #628 from SuperDARN/update_workflow_cdf

Updating compilation test to use new NASA CDF library version

ksterne committed:
Merge pull request #629 from SuperDARN/add_ecdip_ts18_kp

Adding TS18-Kp convection model support for eccentric dipole coordinates

egthomas committed:
Merge branch 'develop' into feature/grid_slant_range

pasha-ponomarenko committed:
Merge pull request #625 from SuperDARN/trim_iq_chnnum

Modifying trim_iq to add support for optional chnnum

ksterne committed:
Merge pull request #618 from SuperDARN/feature/tdiff_multi_freq

Adding tdiff.dat files for multi-frequency calibration results

mtwalach committed:
Merge pull request #616 from SuperDARN/feature/grid_slant_range

Adding new srng field to grid and map files + range/CPID filtering with grid_filter

mtwalach commented:
Hey @ksterne - nice work; how would you like me to test it? Or is it just a review? Depending on what it needs I might need more time.

ksterne commented:
I don't think there's any specific tests aside from trying to use this branch as much as possible for your everyday tasks to make sure nothing breaks or nothing unexpected happens.  I'm open to hear if there's something specifically that we need to test out.  I think there's some people that have been using the develop branch for the last several months or more than a year, so we should be fairly stable.

pasha-ponomarenko reviewed (approved):
I have quickly tested the whole "production cycle" from creating FITACF from RAWACF to generating MAP files and plotting GIF maps. Seems to be working OK.

ksterne committed:
Updating version number

aburrell commented on .github/test_profiles/makelib.darwin:
```diff
@@ -0,0 +1,60 @@
+
+clean:
+	rm -f *.o
+
+
+
+
+
+ 
+
+
+
+
+
```
```suggestion
```

aburrell commented on .readthedocs.yaml:
```diff
@@ -0,0 +1,14 @@
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
```
```suggestion
    python: "3.13"
```

aburrell commented on codebase/analysis/src.bin/aacgm_v2/aacgm_test.1.0/test_aacgm.c:
```diff
@@ -230,7 +232,7 @@ void next(void)
   } while (ch != '\n');
 
-/*  printf("\n\n");*/
+//  printf("\n\n");
```
Recommend going back to c-style commenting instead of using c++ style.

aburrell replied:
Or we can just delete this comment.

ksterne replied:
I don't see a reason to keep the comment in.  It can be added back later if needed.

aburrell commented on codebase/general/src.lib/time.1.7/src/time.c:
```diff
@@ -277,3 +277,48 @@ double TimeStrToEpoch(char *text)
+;+-----------------------------------------------------------------------------
+*/
+
+int dayno(int year, int month, int day, int *diy)
```
This name `dayno` is also frequently used as a variable and a structure element.  I'd recommend renaming this function.

ksterne replied:
While I agree, is this something that will be a lot of work to check all of the places that they function name needs to be changed at?

aburrell replied:
I can convert this to an issue.

aburrell replied:
Done: #633

aburrell reviewed (commented):
I attempted compilation on my linux box, where it failed here:

```
================================================================================
Compiling Binary:/home/aburrell/Git/rst/codebase/analysis/src.bin/aacgm_v2/aacgm_test.1.0
================================================================================
make clean
rm -f *.o
rm -f version.h
rm -f errstr.h
rm -f hlpstr.h
rm -f test_aacgm
make
make.version /data/home/aburrell/Git/rst/codebase/analysis/src.bin/aacgm_v2/aacgm_test.1.0
make.help
cc -fPIC -Wall -pedantic -O3 -D_GNU_SOURCE -D_LINUX  -D_SVGLIB_ -I/home/aburrell/Git/rst/include/base -I/home/aburrell/Git/rst/include/general -I/home/aburrell/Git/rst/include/analysis -c test_aacgm.c    
test_aacgm.c: In function ‘main’:
test_aacgm.c:15:8: warning: unused variable ‘rtp’ [-Wunused-variable]
 double rtp[3];
        ^~~
mkdir -p /home/aburrell/Git/rst/bin
cc -L/home/aburrell/Git/rst/lib -o /home/aburrell/Git/rst/bin/test_aacgm test_aacgm.o -Wl,-Bstatic \
                           -laacgm_v2.1 -ligrf_v2.1 -lmlt_v2.1 -lastalg.1 -lopt.1 -lrtime.1  -Wl,-Bdynamic \
                           -lm   
/home/aburrell/Git/rst/lib/librtime.1.a(time.o): In function `dayno':
time.c:(.text+0xae0): multiple definition of `dayno'
/home/aburrell/Git/rst/lib/libaacgm_v2.1.a(genmag.o):genmag.c:(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [/home/aburrell/Git/rst/build/make/makebin.linux:43: test_aacgm] Error 1
Compilation Aborted.
```

aburrell commented on codebase/superdarn/src.idl/lib/main.1.25/cnvmap.pro:
```diff
@@ -228,7 +228,7 @@ function CnvMapRead,unit,prm,stvec,gvec,mvec,coef,bvec
            
 
   arrtype=[2,2,2,4,2,2,2,4,4,2,4,4,4,4,4,4,4,4, $
-           4,4,4,2,2,3,4,4,4,4,4,4,8,8,8,8,4,4,4,4,4,4]
+           4,4,4,4,2,2,3,4,4,4,4,4,4,8,8,8,8,4,4,4,4,4,4]
```
Would be useful to have a comment here shortly saying what this array is holding.

ksterne replied:
I think we'll leave this one for resolving in #631

aburrell commented on codebase/superdarn/src.idl/lib/main.1.25/grd.pro:
```diff
@@ -198,13 +199,13 @@ function GridRead,unit,prm,stvec,gvec
            'vector.wdt.median','vector.wdt.sd']
 
-  arrtype=[2,2,2,4,2,2,2,4,4,2,4,4,4,4,4,4,4,4,4,4,4,2,2,3,4,4,4,4,4,4]
+  arrtype=[2,2,2,4,2,2,2,4,4,2,4,4,4,4,4,4,4,4,4,4,4,4,2,2,3,4,4,4,4,4,4]
```
Also recommend adding that comment here (appears to be the same type of array).

ksterne replied:
Similiarly, I think we'll leave this one to resolve in #631

aburrell commented on codebase/superdarn/src.lib/tk/iq.1.7/src/iqwrite.c:
```diff
@@ -61,7 +61,7 @@ int IQEncode(struct DataMap *ptr,struct IQ *iq,
   bnum=iq->tbadtr*2;
 
- snum=iq->seqnum*iq->chnnum*iq->smpnum*2*2;   /*2 for main and back  2  to convert to 16 bit */  
+  snum=iq->seqnum*iq->chnnum*iq->smpnum*2;
```
Recommend adding a comment above that describes what's being done that is different than before.

ksterne replied:
@egthomas, is this something that you worked on and would be able to provide a comment here?

ksterne commented:
Thanks for the checks and reviews so far.  I won't be able to address them today, but early next week.  For now I wanted to post the draft author list as:

SuperDARN Data Analysis Working Group; Thomas, E.G.; Burrell, A.G.; Ponomarenko, P.V.; Bland. E.C.; Sterne, K.T.; Rohel, R.; Billet, D.D.; Reimer, A.S.; Schmidt, M.T.; Walach, M.-T.

This follows the authorship guidelines from a few years ago: https://superdarn.github.io/dawg/documents/rst-authorship-criteria/ 

@RemingtonRohel, let me know if I've credited you correctly or if there's a better way that you'd like to be credited with authorship.

aburrell reviewed (commented):
Found some places where comments would help with maintenance.

ksterne committed:
Draft Zenodo author list

ksterne committed:
Updating contributing authors

ksterne commented:
On an Ubuntu 22 install, I'm getting the same thing as @aburrell where the compilation is failing as the function `dayno` is repeated between librtime.1.a and the aacgm_test.1.0.  It looks like we encountered this in #607 and the fix is to remove the `lib/` directory and then recompile.  Should we add something to the `make.code` script to remove the top level library directory to make this more user friendly?

RemingtonRohel commented:
@ksterne credit looks good to me, thanks!

aburrell commented:
> Should we add something to the make.code script to remove the top level library directory to make this more user friendly?

@ksterne yes please!

ksterne committed:
Update start date from JJ Zhang

ksterne committed:
Update start dates from JJ Zhang

ksterne committed:
Enabling status flag for new radars

ksterne commented:
I've heard from Jiaojiao Zhang about updates to the start date for the new Meridian Project radars and have added those to the corresponding `radar.dat` and `hdw.dat` files.  Also just sent a reminder to the PIs to review their hdw.dat files for any updates that might be needed with a note that we're looking to make a release on the 4th.

egthomas commented:
> I've heard from Jiaojiao Zhang about updates to the start date for the new Meridian Project radars and have added those to the corresponding `radar.dat` and `hdw.dat` files. Also just sent a reminder to the PIs to review their hdw.dat files for any updates that might be needed with a note that we're looking to make a release on the 4th.

@ksterne shouldn't the radar status flag also be updated from `0` to `1` in both the `radar.dat` and `hdw.dat` files?

> Thanks for the checks and reviews so far. I won't be able to address them today, but early next week. For now I wanted to post the draft author list as:
>
> SuperDARN Data Analysis Working Group; Thomas, E.G.; Burrell, A.G.; Ponomarenko, P.V.; Bland. E.C.; Sterne, K.T.; Rohel, R.; Billet, D.D.; Reimer, A.S.; Schmidt, M.T.; Walach, M.-T.
>
> This follows the authorship guidelines from a few years ago: https://superdarn.github.io/dawg/documents/rst-authorship-criteria/

This list is missing `Shepherd, S. G.` who wrote the code for the latest AACGM-v2 / IGRF-v2 update and derived the revised coefficients.  Since the capability for those new coordinate transformations is what finally prompted a new RST release, that seems like it should warrant inclusion in the author list.

ksterne commented:
@egthomas, that's a good question about the radar status flag.  I don't think the data is being distributed yet, so there might be reasons to keep it at 0.  I'll follow up with Jiaojiao though.

For the authorship, I followed the guidelines we drew up a few years ago with one of the points of contributions that do not qualify authorship is being the author of a package dependency, or an external library that is distributed with the software.  AACGM-v2 is released in a separate location by Simon than this repo and didn't change any of the code in this repo.

I'd like to hear others thoughts on the interpretation of the guidelines and including Simon as an author.

mtwalach committed:
Update .github/test_profiles/makelib.darwin

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

aburrell commented:
I think Simon should be included as an author, since RST incorporates his code locally and doesn't install it externally (like it does with CDF).

mtwalach reviewed (commented):
Looks mostly good, though I concur with @aburrell on the points she's made. 
Worth noting that I have not been able to reinstall it on my Mac yet.

mtwalach commented:
..and I also agree that we should add Simon to the list of authors since there is some of his code in there and just having AACGM externally to RST simply wouldn't work as it is set up. 😃

ksterne commented:
@egthomas, I've heard back from Jiaojiao that we're good to go ahead and enable the status flag for their new radars as they should be distributing the data sometime soon.  Maybe double check I didn't miss something with 49b1756.

egthomas commented:
> On an Ubuntu 22 install, I'm getting the same thing as @aburrell where the compilation is failing as the function `dayno` is repeated between librtime.1.a and the aacgm_test.1.0. It looks like we encountered this in #607 and the fix is to remove the `lib/` directory and then recompile. Should we add something to the `make.code` script to remove the top level library directory to make this more user friendly?

While I understand the motivation, by deleting the top-level `lib` directory on each execution of `make.code` we would lose the ability to re-compile individual pieces of the RST (e.g., `make.code bin make_fit`).  So I would not be in favor of making such a change.

Maybe a simpler alternative would be adding a disclaimer to the installation instructions that we cannot guarantee compatibility between releases when pulling from github vs making a fresh installation from eg Zenodo.  In fact, we were asked several years ago to link from our website to the latest release on the Zenodo archive rather than directly to the release on the github repository.

aburrell commented:
What about adding a flag that would enable a clean installation as an option?  Then the default behaviour would remain the same, but we would simplify the clean up needed for many users.  The instructions are already a lot.

ksterne committed:
Add author after feedback

ksterne committed:
Removing commented line

aburrell commented:
Created #634 to contain discussion/work on the clean install options.

ksterne committed:
Changing author list after 20250414 DAWG meeting

ksterne committed:
Update .readthedocs.yaml

Co-authored-by: Angeline Burrell <aburrell@users.noreply.github.com>

egthomas commented on .zenodo.json:
```diff
@@ -10,46 +10,46 @@
+        },
+        {
+            "affiliation": "Dartmouth College",
+            "name": Shepherd, S.G.",
```
```suggestion
            "name": "Shepherd, S.G.",
```

there still seems to be a formatting error here

ksterne replied:
Good catch, I couldn't figure out why github was highlighting it strangely!  Sometimes the obvious things blow past me.

ksterne committed:
Update .zenodo.json

Co-authored-by: Evan Thomas <evan.g.thomas@dartmouth.edu>

ksterne commented:
Hello all, during the Data Analysis Working Group meeting today, we had some more discussion on the authorship for this release.  Two main points that I've committed already:
1. Removed Ashton Reimer as think the contribution that was made was a while ago.
2. Moved Simon to earlier in the co-author list with being out of the alphabetical section and to the major contributors.

We wanted to get others opinion on this since not everyone could make the meeting today.  In particular, @mtwalach, @aburrell do you agree with the placement of Simon on the list?

Also, we worked through the release notes which have been copied to the original posting here for reference.

mtwalach reviewed (approved):
Yeah, I think that's absolutely appropriate and within the spirit of the WG.

mtwalach commented:
Re: release notes, one suggestion: 
I wonder if you could make it a tiny bit more concise e.g. 
combine the bullet points "Updates to installation instructions for Mac OS" and "Various documentation improvements" 
Into: "Various documentation improvements (e.g. updated installation instructions for Mac OS)". 

That's all I spotted. 👍

aburrell commented on .github/test_profiles/profile.bash.darwin:
```diff
@@ -0,0 +1,16 @@
+export SYSTEM="darwin"
+export VISUAL="vim"
+export EDITOR="$VISUAL"
+export PATH="${PATH}:/opt/homebrew/bin:/opt/local/bin:/usr/bin/:/usr/local/bin:${HOME}/bin:${HOME}/script"
```
Same as above comment.

aburrell reviewed (approved):
I don't have a strong opinion on author order this go through, just author inclusion.  If other folks are happy, so am I.  I'm good with this release and am pleased with the issues created to capture the small, identified improvements that can be made.
